src/shared/components/events-list/index.js

```diff
@@ -2,134 +2,123 @@
 // You can request only displaying events of past or future
 // with the `timeline` prop
 
-/* eslint-disable max-len */
-
-import React, { Component } from 'react';
+import React, { PropTypes } from 'react';
 import styles from './style.css';
 
 import EventImage from '../event-image';
 import { imageAssetsEndpoint } from '../../config';
 import DateBubble from '../date-bubble';
 import HR from '../hr';
 import { Grid, Cell } from '../grid';
-import classNames from 'classnames';
-import icons from '../icons/style.css';
 
 import TagsList from '../tags-list';
 import EventLinksList from '../event-links-list';
+import EventsTimelineTitle from '../events-timeline-title';
+import EventTitle from '../event-title';
 import { eventHref } from '../../util/event';
 import { splitEvents } from '../../util/split-events';
 
-export default class EventsList extends Component {
-  static propTypes = {
-    events: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
-    timeline: React.PropTypes.oneOf(['past', 'future', 'today']),
-  };
+const EventsList = ({
```

Comment:
Lots of new logic in this component, but no new tests. Do the existing ones cover the new paths?